tests/test_custom_checkers/test_top_level_code_checker.py

```diff
@@ -8,195 +9,113 @@ class TestTopLevelCodeChecker(pylint.testutils.CheckerTestCase):
-        """
-        mod = astroid.parse(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="forbidden-top-level-code", node=mod.body[0], args=2
+            (
+                """
+                name = "George"
+                """,
+                2,
             ),
-            ignore_position=True,
-        ):
-            self.checker.visit_module(mod)
-
-    def test_no_message_type_alias_assignment(self):
-        """Top level type alias assignment allowed, no message."""
-        src = """
-        MyType = list[list[list[int]]]
-        """
-        mod = astroid.parse(src)
-        with self.assertNoMessages():
-            self.checker.visit_module(mod)
-
-    def test_message_type_alias_assignment(self):
-        """Top level type alias assignment not allowed, raises a message."""
-        src = """
-        TypeName = list[int]
-        """
-        mod = astroid.parse(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="forbidden-top-level-code", node=mod.body[0], args=2
+            (
+                """
+                TypeName = list[int]
+                """,
+                2,
             ),
-            ignore_position=True,
-        ):
-            self.checker.visit_module(mod)
-
-    def test_message_regular_assignment_unpacking(self):
-        """Top level regular unpacking assignment not allowed, raises a message."""
-        src = """
-        name, CONST = "George", 3
-        """
-        mod = astroid.parse(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="forbidden-top-level-code", node=mod.body[0], args=2
+            (
+                """
+                name, CONST = "George", 3
+                """,
+                2,
             ),
-            ignore_position=True,
-        ):
-            self.checker.visit_module(mod)
-
-    def test_message_regular_assignment_starred(self):
-        """Top level regular assignment with a starred target not allowed, raises a message."""
-        src = """
-        NAME, *nums = ["George", 3, 4]
-        """
-        mod = astroid.parse(src)
-        with self.assertAddsMessages(
-            pylint.testutils.MessageTest(
-                msg_id="forbidden-top-level-code", node=mod.body[0], args=2
+            (
+                """
+                NAME, *nums = ["George", 3, 4]
+                """,
+                2,
             ),
+            (
+                """
+                class X:
+                    a = 5
+                Y = X()
+                Y.a = 6
+                """,
+                5,
+            ),
+        ],
+    )
+    def test_forbidden_top_level_code(self, src, lineno):
+        """Cases that must raise forbidden-top-level-code"""
+        mod = astroid.parse(src)
+        # Get the correct node: last body element for the attribute one
```

Comment:
This code is a bit awkward. You can instead assume that the _last_ statement in the body is `node` associated with the error. Please add this assumption to the function docstring.